pkg/controller/utils/clusterdeploymentbuilder.go

```diff
@@ -0,0 +1,111 @@
+package utils
+
+import (
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+type clusterDeploymentChecksumFunc func(clusterDeployment *hivev1.ClusterDeployment) (string, error)
+
+// ClusterDeploymentBuilder is a utility to build ClusterDeployments given specific requirements.
+// It's especialy useful in cases where simply defining a ClusterDeployment struct isn't sufficient
+// (e.g. when declaring a variable that needs to checksum attribute set.)
+type ClusterDeploymentBuilder struct {
+	ClusterDeployment *hivev1.ClusterDeployment
+	ChecksumFunc      clusterDeploymentChecksumFunc
```

Comment:
Why not make this a parameter to `WithBackupChecksum` rather than including it in the builder type?

Reply:
It was mostly so that the 90% correct objects could specify it once and then all of the subsequent calls wouldn't need it specified.

I'm fine changing it if you feel strongly it should be the other way around.

Reply:
I'll reserve judgement on this until I see how it plays out with the switch to functional options.